src/general/smoothing_kernels.jl

```diff
@@ -676,3 +676,71 @@ end
 
 @inline normalization_factor(::SpikyKernel{2}, h) = 10 / (pi * h^2)
 @inline normalization_factor(::SpikyKernel{3}, h) = 15 / (pi * h^3)
+
+@doc raw"""
+    LaguerreGaussKernel{NDIMS}()
+
+Truncated Laguerre–Gauss kernel (fourth-order smoothing) as defined in
+[Wang2024](@cite). Its radial form uses
+`s = r/h` and is truncated at `s = 2` (compact support `2h`):
+
+```math
+W(r,h) = \frac{1}{h^d} \, w(s), \quad s=\frac{r}{h}, \quad
+w(s) = \alpha_d \left(1 - \frac{s^2}{2} + \frac{s^4}{6}\right) e^{-s^2},
+\quad 0 \le s \le 2,
+with normalization constants
+α₁ = 8/(5√π), α₂ = 3/π, α₃ = 8/π^{3/2}.
+
+Recommended practical choice from the paper: use h ≈ 1.3Δx and the same
+cut-off as Wendland (2h) for comparable cost. Negative lobes enforce the
+vanishing second moment (4th-order smoothing) while remaining stable in
+Eulerian / total Lagrangian SPH with relaxed particles.
+
+For general information and usage see [Smoothing Kernels](@ref smoothing_kernel).
+"""
+struct LaguerreGaussKernel{NDIMS} <: AbstractSmoothingKernel{NDIMS} end
+
+@fastpow @muladd @inline function kernel(kernel::LaguerreGaussKernel, r::Real, h)
+    s = r / h
+    # polynomial part: 1 - s^2/2 + s^4/6
+    poly = 1 - s^2 / 2 + s^4 / 6
+    val = normalization_factor(kernel, h) * poly * exp(-s^2)
+    # zero out for s ≥ 2
+    return ifelse(s < 2, val, zero(val))
+end
+
+@muladd @inline function kernel_deriv(kernel::LaguerreGaussKernel, r::Real, h)
```

Comment:
```suggestion
@inline function kernel_deriv(kernel::LaguerreGaussKernel, r::Real, h)
```
There is no muladd used anywhere in this function. Same for the one above.